src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Configuration/ConfigurationProjectConfigurationDimensionProvider.cs

```diff
@@ -0,0 +1,107 @@
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.VS.Editor;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Configuration
+{
+    /// <summary>
+    /// Provides 'Configuration' project configuration dimension and values.
+    /// </summary>
+    [Export(typeof(IProjectConfigurationDimensionsProvider2))]
+    [AppliesTo(ProjectCapabilities.ProjectConfigurationsDeclaredDimensions)]
+    [Order(int.MaxValue)]
+    internal class ConfigurationProjectConfigurationDimensionProvider : BaseProjectConfigurationDimensionProvider
+    {
+        [ImportingConstructor]
+        public ConfigurationProjectConfigurationDimensionProvider(IProjectXmlAccessor projectXmlAccessor)
+            : base(projectXmlAccessor, ConfigurationGeneral.ConfigurationProperty, "Configurations")
+        {
+        }
+
+        /// <summary>
+        /// Modifies the project when there's a configuration change.
+        /// </summary>
+        /// <param name="args">Information about the configuration dimension value change.</param>
+        /// <returns>A task for the async operation.</returns>
+        public override async Task OnDimensionValueChangedAsync(ProjectConfigurationDimensionValueChangedEventArgs args)
+        {
+            if (args.DimensionName == _dimensionName)
+            {
+                if (args.Stage == ChangeEventStage.Before)
+                {
+                    switch (args.Change)
+                    {
+                        case ConfigurationDimensionChange.Add:
+                            await OnConfigurationAdded(args.Project, args.DimensionValue).ConfigureAwait(true);
+                            break;
+                        case ConfigurationDimensionChange.Delete:
+                            await OnConfigurationRemoved(args.Project, args.DimensionValue).ConfigureAwait(true);
+                            break;
+                        case ConfigurationDimensionChange.Rename:
+                            // Need to wait until the core rename changes happen before renaming the property.
+                            break;
+                    }
+                }
+                else if (args.Stage == ChangeEventStage.After)
+                {
+                    // Only change that needs to be handled here is renaming configurations which needs to happen after all
+                    // of the core changes to rename existing conditions have executed.
+                    if (args.Change == ConfigurationDimensionChange.Rename)
+                    {
+                        await OnConfigurationRenamed(args.Project, args.OldDimensionValue, args.DimensionValue).ConfigureAwait(true);
+                    }
+
+                    // await _projectXmlAccessor.ExecuteInWriteLock((msbuildProject) => msbuildProject.Save()).ConfigureAwait(true);
```

Comment:
Oops, I was trying to see if I could work around another issue and forgot to delete this.